docs/guide/controller-programming.md

```diff
@@ -247,60 +563,265 @@ Instead it stores the data locally and the data is effectively sent when the `wb
+%tab-end
+
+%tab "C++"
+```cpp
+while (robot->step(40) != -1) {
+  double d1 = robot->getDistanceSensor(sensor);
+  if (robot->step(40) == -1)
+    break;
+  double d2 = robot->getDistanceSensor(sensor);
+  if (d2 > d1)
+    avoidCollision();
+}
+```
+%tab-end
+
+%tab "Python"
+```python
+while robot.step(40) != -1:
+  d1 = robot.getDistanceSensor(sensor)
+  if robot.step(40) == -1:
+    break
+  d2 = robot.getDistanceSensor(sensor)
+  if d2 > d1:
+    avoidCollision()
+```
+%tab-end
+
+%tab "Java"
+```java
+while (robot.step(40) != -1) {
+  double d1 = robot.getDistanceSensor(sensor);
+  if (robot.step(40) == -1)
+    break;
+  double d2 = robot.getDistanceSensor(sensor);
+  if (d2 > d1)
+    avoidCollision();
+}
+```
+%tab-end
+
+%tab "MATLAB"
+```MATLAB
+while wb_robot_step(40) ~= -1
+  d1 = wb_distance_sensor_get_value(sensor);
+  if wb_robot_step(40) == -1
+    break
+  end
+  d2 = wb_distance_sensor_get_value(sensor);
+  if d2 > d1
+    avoidCollision();
+  end
+end
+```
+%tab-end
+%end
 
 However, the generally recommended approach is to have a single `wb_robot_step` function call in the main control loop, and to use it to update all the sensors and actuators simultaneously, like this:
 
+%tab-component "language"
+%tab "C"
 ```c
-while (wb_robot_step(TIME_STEP) != -1) {
+while (wb_robot_step(40) != -1) {
   readSensors();
   actuateMotors();
 }
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+while (robot->step(40) != -1) {
+  readSensors();
+  actuateMotors();
+}
+```
+%tab-end
+
+%tab "Python"
+```python
+while robot.step(40) != -1:
+  readSensors()
+  actuateMotors()
+```
+%tab-end
+
+%tab "Java"
+```java
+while (robot.step(40) != -1) {
+  readSensors();
+  actuateMotors();
+}
+```
+%tab-end
+
+%tab "MATLAB"
+```MATLAB
+while robot.step(40) != -1
+  readSensors();
+  actuateMotors();
+end
+```
+%tab-end
+%end
 
 Note that it is important to call the `wb_robot_step` function at the beginning of the loop, in order to make sure that the sensors already have valid values prior to entering the `readSensors` function.
 Otherwise the sensors will have undefined values during the first iteration of the loop, hence, the following is not a good example:
+Note the following snippets are only translated to languages that support builtin `do...while` statements. The point is to insist that we should always rely on `wb_robot_step` _before_ making any measure.
 
+%tab-component "language"
+%tab "C"
 ```c
 do {
   readSensors(); // warning: sensor values are undefined on the first iteration
   actuateMotors();
-} while (wb_robot_step(TIME_STEP) != -1);
+} while (wb_robot_step(40) != -1);
+```
+%tab-end
+
+%tab "C++"
+```cpp
+do {
+  readSensors(); // warning: sensor values are undefined on the first iteration
+  actuateMotors();
+} while (robot->step(40) != -1);
 ```
+%tab-end
+
+%tab "Java"
+```java
+do {
+  readSensors(); // warning: sensor values are undefined on the first iteration
+  actuateMotors();
+} while (robot.step(40) != -1);
+```
+%tab-end
+%end
```

Comment:
Is it normal that Python and MATLAB are missing here?

Reply:
Yes, it is stated above that only languages natively supporting the `do...while` statement are demonstrated, this is also completed by a sentence afterwards which emphasis on the need to call `wb_robot_step()` function :)